packages/flutter/lib/src/material/chip.dart

```diff
@@ -245,6 +245,38 @@ abstract class DeletableChipAttributes {
   String get deleteButtonTooltipMessage;
 }
 
+/// An interface for material design chips that can have check marks.
+///
+/// The defaults mentioned in the documentation for each attribute are what
+/// the implementing classes typically use for defaults (but this class doesn't
+/// provide or enforce them).
+///
+/// See also:
+///
+///  * [InputChip], a chip that represents a complex piece of information, such
+///    as an entity (person, place, or thing) or conversational text, in a
+///    compact form.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * <https://material.io/design/components/chips.html>
+abstract class CheckmarkableChipAttributes {
```

Comment:
I know there's already a pattern of `FooChipAttributes`, but my personal preference would be to just add these parameters individually on any "CheckMarkable" chip. Then the chip types can pipe the params to `RawMaterialChip`. 

I guess there isn't really a con to having this though. Let's see what others think!

Reply:
I did this mainly for consistency.  A new interface had to be added because although ChoiceChip, InputChip, and FilterChip are all selectable, ChoiceChip is not checkmarkable.

It is nice to use the interfaces because the documentation for each filed is inherently centralized (of course macros could do this too)